impersonate/middleware.py

```diff
@@ -1,26 +1,55 @@
+from django.utils.functional import empty, SimpleLazyObject
 from .helpers import User, check_allow_for_uri, check_allow_for_user
 
 
-class ImpersonateMiddleware(object):
+def apply_impersonate(request):
+    request.user.is_impersonate = False
+    request.__dict__['impersonator'] = None
```

Comment:
this and the other impersonation __dict__ access should be turned into normal assignments